Rubberduck.CodeAnalysis/Inspections/Concrete/EmptyMethodInspection.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing.Common;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Resources.Inspections;
+using Rubberduck.Parsing.VBA;
+using System.Collections.Generic;
+using System.Linq;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Resources.Experimentals;
+using static Rubberduck.Parsing.Grammar.VBAParser;
+
+namespace Rubberduck.Inspections.Concrete
+{
+
+    /// <summary>
+    /// Identifies empty module member blocks.
+    /// </summary>
+    /// <why>
+    /// Methods containing no executable statements are misleading as they appear to be doing something which they actually don't.
+    /// This might be the result of delaying the actual implementation for a later stage of development, and then forgetting all about that.
+    /// </why>
+    /// <example hasResults="true">
+    /// <![CDATA[
+    /// Sub Foo()
+    ///     ' ...
+    /// End Sub
+    /// ]]>
+    /// </example>
+    /// <example hasResults="false">
+    /// <![CDATA[
+    /// Sub Foo()
+    ///     MsgBox "?"
+    /// End Sub
+    /// ]]>
+    /// </example>
+    [Experimental(nameof(ExperimentalNames.EmptyBlockInspections))]
+    internal class EmptyMethodInspection : InspectionBase// EmptyBlockInspectionBase
+    {
+        public EmptyMethodInspection(RubberduckParserState state)
+            : base(state) { }
+
+        protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
+        {
+            // Exclude empty members in user interfaces, as long as all members of the interface are empty,
+            // since some VB users might use concrete user defined classes as interfaces,
+            // while RD marks them as interfaces all the same.
+
+            return UserDeclarations.OfType<ModuleBodyElementDeclaration>()
+                .Where(bodyElement => !BlockContainsExecutableStatements(bodyElement.Block))
+                .GroupBy(bodyElement => bodyElement.ComponentName)
+                // Exclude results from user interfaces
+                .Where(bodyElements => !State.DeclarationFinder.FindAllUserInterfaces()
```

Comment:
Finding all user interfaces for every `bodyElements` could be made slightly more efficient by caching them in a local hashset, regardless of whether the finder caches them or not (IIRC it does).